llvm/lib/MC/MCPseudoProbe.cpp

```diff
@@ -605,13 +616,20 @@ bool MCPseudoProbeDecoder::buildAddress2ProbeMap(
     TopLevelFuncs += !Discard;
   }
   assert(Data == End && "Have unprocessed data in pseudo_probe section");
+  PseudoProbeVec.reserve(ProbeCount);
+  InlineTreeVec.reserve(InlinedCount);
```

Comment:
Since this assumes the vector's reserved size will be equal to the amount it will be used. How about adding assertions at the end of function. like `assert(InlineTreeVec.size() == InlinedCount && ...)` this is for checking any accidental re-allocation/extension of the vector during the build the map.